homeassistant/components/input_datetime/__init__.py

```diff
@@ -120,13 +122,18 @@ def __init__(self, object_id, name, has_date, has_time, icon, initial):
             if old_state is not None:
                 restore_val = old_state.state
 
-        if restore_val is not None:
-            if not self.has_date:
-                self._current_datetime = dt_util.parse_time(restore_val)
-            elif not self.has_time:
-                self._current_datetime = dt_util.parse_date(restore_val)
-            else:
-                self._current_datetime = dt_util.parse_datetime(restore_val)
+        if not self.has_date:
```

Comment:
Instead of all these changes, just add 2 lines above the old code:

```
if not restore_val:
    restore_val = DEFAULT_VALUE
```

Reply:
Fixed in latest commit

Reply:
This won't work, as parse_date and parse_time expect each to get a string of the form "%Y-%m-%d" or "%H:%m:%s" respectively.
Please advise.

Reply:
oh right, fine like this.